src/Client/ClientBase.cpp

```diff
@@ -2372,6 +2373,32 @@ void ClientBase::runInteractive()
     if (print_time_to_stderr)
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "time option could be specified only in non-interactive mode");
 
+    if (stdout_is_a_tty && FormatFactory::instance().checkIfOutputFormatIsBinary(format))
```

Comment:
1. Missing tests.
2. This check will happen at the client startup and only in the interactive mode rather than for a query.

Check what happens if a user issued multiple queries, e.g. `SELECT 1 FORMAT TSV`, `SELECT 1 FORMAT Native`.